proxy/src/main/java/com/velocitypowered/proxy/connection/client/ClientConfigSessionHandler.java

```diff
@@ -141,14 +143,17 @@ public boolean handle(PingIdentifyPacket packet) {
 
   @Override
   public boolean handle(KnownPacksPacket packet) {
-    callConfigurationEvent().thenRun(() -> {
-      player.getConnectionInFlightOrConnectedServer().ensureConnected().write(packet);
-    }).exceptionally(ex -> {
-      logger.error("Error forwarding known packs response to backend:", ex);
-      return null;
-    });
+    if (player.getConnectionInFlight() != null) {
```

Comment:
So that IF could just be moved in front of 

>        player.getConnectionInFlight().ensureConnected().write(packet);

tho... it might affect plugins listening to config event?

Reply:
that would prevent plugins from being able to interact with reconfiguration while on a server, this block basically needs reverting

Reply:
Well, does the latest commit work or not, because, that looks pretty on par with what I'd expect from this?

Reply:
I wouldn't be calling handleGeneric, however, we should ideally be handling all packets inside of this phase, as that we, any packet we don't know about is going to be a modded packet, which is where the special handling gets into play here